### DISCUSSION TITLE
v8Flags.setMaxHeapSize() is this for java heap memory or native memory?

### DISCUSSION THREAD

**Discussion by RakeshRaparthi** [caoccao/Javet#429] [Q&A]
I am using javet lib in java application, I just want o know v8Flags.setMaxHeapSize() is for heap memory or native memory? little bit confusing...
i believe this memory is sharedMemory for all engines. could anyone please clarify?
i have seen crash for big js payload like 30 mb having setMaxHeapSize as 4gb. what are recommend options.?

**Comment by caoccao** [comment]
Please refer to the official doc , or leverage Node.js help for the answer.

```sh
node --v8-options | grep 'max-heap'
  --max-heap-size (max size of the heap (in Mbytes) both max_semi_space_size and max_old_space_size take precedence. All three flags cannot be specified at the same time.)
        type: size_t  default: --max-heap-size=0
```

If you still face issues, could you leave a repo with the issues?

**Comment by RakeshRaparthi** [comment]
I could n't find your doc which says using native memory or heap memory.
could you please share.

**Reply by caoccao** [reply]
It is not managed by JVM.

**Reply by RakeshRaparthi** [reply]
is there any way to some control in java, we could catch it maybe we can retry?

**Reply by caoccao** [reply]
There might be some ways of doing so, but those ways haven't been implemented anywhere. Not even in Chrome or Node.js. They just crash anyway. Please refer to this [blog](https://medium.com/towardsdev/break-serverless-in-node-js-easily-c73b8c9ca13) for detail.
